File: Samples/swUSB/Termo 2 sensors/PC/C#/USB_termo_2/Form1.cs
using System.Diagnostics;
using System;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data;
using System.Collections.Generic;


namespace USB_termo_2
{
    public partial class Form1 : Form
    {
        int vid = 0xaaaa;
        int pid = 0xef04;
        short ver = -1;
        short ind = 0;
        int Handless = 0;
        byte[] Buffer = new byte[5];
        int State;
        int res;


        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            //При открытии формы открыть девайс 
            Handless = v_usb.HID_OpenDevice(pid, vid, ver, ind); 

        }

        private void timer1_Tick(object sender, EventArgs e)
        {
 
        State = v_usb.HID_DeviceTest(pid, vid, ver, ind);

        if (State == 0) //если устройство отключено
            {
            label1.Text = "Девайс отключен";
            label2.Text = "Девайс отключен";
            Buffer[1] = 0;
            Buffer[2] = 0;
            Buffer[3] = 0;
            Buffer[4] = 0;
            }
        else    //если устройство подключено, то вычисляем температуру
            {
                Handless = v_usb.HID_OpenDevice(pid, vid, ver, ind);
                res = v_usb.HID_ReadDevice(Handless, Buffer, Buffer.Length);



            //######################################################################################
            // ВЫЧИСЛЕНИЕ ТЕМПЕРАТУРЫ ИЗ ПРИНЯТЫХ БАЙТОВ И ОТОБРАЖЕНИЕ В НЕОБХОДИМОМ ФОРМАТЕ
                double zamer;
                string  temperatura;

                if (Buffer[1] == 254 || Buffer[2] == 254)
                {
                    label1.ForeColor = Color.Red;
                    label1.Text = "No datchik";
                }
                else
                {
                    zamer = Convert.ToDouble(Buffer[1]) / 16 + Convert.ToDouble(Buffer[2])* 16;
                    if (zamer < 150 && zamer > -50)
                    {
                        temperatura = Strings.Format(zamer, ".0");
                        label1.ForeColor = Color.Black;
                        label1.Text = temperatura + " °C";
                    }
                }
         
            if (Buffer[3] == 254 || Buffer[4] == 254) 
            {
                label2.ForeColor = Color.Red;
                label2.Text = "No datchik";
            }
            else
            {
                zamer = Convert.ToDouble(Buffer[3]) / 16 + Convert.ToDouble(Buffer[4]) * 16;
                if (zamer < 150 && zamer > -50) 
                {
                    temperatura = Strings.Format(zamer, ".0");
                    label2.ForeColor = Color.Black;
                    label2.Text = temperatura + " °C";
                }
            }
            }

            //######################################################################################
            

        }

        private void Form1_FormClosed(object sender, FormClosedEventArgs e)
        {
             //При закрытии формы закрыть девайс
            res = v_usb.HID_CloseDevice(Handless);

        }
    }
}

File: Samples/Samples/swUSB/Termo 2 sensors/PC/VBNET/WindowsApplication1/Form1.vb
Public Class Form1

    Dim vid As Integer = &HAAAA '&H16C0 '43690
    Dim pid As Integer = &HEF04 '&H5DF '61188
    Dim ver As Short = -1
    Dim ind As Short = 0
    Dim Handless, res As Integer
    Dim Buffer(4) As Byte

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        'При открытии формы открыть девайс
        Handless = HID_OpenDevice(pid, vid, ver, ind)
    End Sub

    Private Sub Form1_FormClosed(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
        'При закрытии формы закрыть девайс
        res = HID_CloseDevice(Handle)
    End Sub

    Private Sub Timer1_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Timer1.Tick
        ' ВЫЧИСЛЕНИЕ ТЕМПЕРАТУРЫ ИЗ ДВУХ ПРИНЯТЫХ БАЙТОВ И ОТОБРАЖЕНИЕ В НЕОБХОДИМОМ ФОРМАТЕ
        Dim zamer As Decimal
        Dim temperatura
        Dim State

        State = HID_DeviceTest(pid, vid, ver, ind)

        If State = 0 Then 'если устройство отключено
            Label1.Text = "Девайс отключен"
            Buffer(1) = 0
            Buffer(2) = 0
        Else    'если устройство подключено, то вычисляем температуру
            Handless = HID_OpenDevice(pid, vid, ver, ind)
            res = HID_ReadDevice(Handless, Buffer, Buffer.Length)
            'zamer = Buffer(1) / 16 + Buffer(2) * 16
            'If zamer < 150 And zamer > -50 Then
            '    temperatura = Format(zamer, ".0")
            '    Label1.Text = temperatura & " °C"
            'End If


            '######################################################################################
            ' ВЫЧИСЛЕНИЕ ТЕМПЕРАТУРЫ ИЗ ДВУХ ПРИНЯТЫХ БАЙТОВ И ОТОБРАЖЕНИЕ В НЕОБХОДИМОМ ФОРМАТЕ
            If Buffer(1) = 254 And Buffer(2) = 254 Then
                Label1.ForeColor = Color.Red
                Label1.Text = "No datchik"
            Else
                zamer = Buffer(1) / 16 + Buffer(2) * 16
                If zamer < 150 And zamer > -50 Then
                    temperatura = Format(zamer, ".0")
                    Label1.ForeColor = Color.Black
                    Label1.Text = temperatura & " °C"
                End If

            End If

            If Buffer(3) = 254 And Buffer(4) = 254 Then
                Label2.ForeColor = Color.Red
                Label2.Text = "No datchik"
            Else
                zamer = Buffer(3) / 16 + Buffer(4) * 16
                If zamer < 150 And zamer > -50 Then
                    temperatura = Format(zamer, ".0")
                    Label2.ForeColor = Color.Black
                    Label2.Text = temperatura & " °C"
                End If

            End If

            '######################################################################################





        End If

    End Sub

End Class
